### PR TITLE
test: migrate tests to HostProcess pods

### LINKED ISSUE #227: Update the integration tests to run in a HostProcess pod

### PR DESCRIPTION
**Pull request by alexander-ding**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**:
As we migrate to a library model, the tests need to more closely mimic the runtime environment of the code, which is in HostProcess containers. Thus, this PR updates the tests to spin up a pod running HostProcess containers to execute the tests, instead of directly running the test binary on the Windows node.

**Which issue(s) this PR fixes**:

Fixes #227

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @mauriciopoppe

### CODE REVIEW

File: scripts/integration-test.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+metadata:
+  name: integration-test
+  labels:
+    app: integration-test
+spec:
+  nodeName: windows_node # to be replaced by sed
+  securityContext:
+    windowsOptions:
+      hostProcess: true
+      runAsUserName: "NT AUTHORITY\\SYSTEM"
```
Comment (mauriciopoppe) [inline scripts/integration-test.yaml:12]: let's experiment here, could you try the user with the lowest privilege first?

File: scripts/utils.sh [hunk 2]
```diff
@@ -33,12 +37,13 @@ sync_csi_proxy_integration_tests() {
 run_csi_proxy_integration_tests() {
   echo "Run integration tests"
   local ps1=$(cat << 'EOF'
-  "& {
-    $ErrorActionPreference = \"Stop\";
-    .\integrationtests.test.exe --test.v
-  }"
+    $ErrorActionPreference = "Stop";
```
Comment (mauriciopoppe) [inline scripts/utils.sh:40]: did it fail when it was in its own scope e.g. when inside `&{ }`

Comment (alexander-ding) [reply scripts/utils.sh:40]: The error code doesn't propagate correctly when in a subshell. We want the overall command to fail if the test fails.

File: scripts/integration-test.yaml (hunk 1 above)
Comment (alexander-ding) [reply scripts/integration-test.yaml:12]: In my previous experiments, anything less than SYSTEM hasn't worked. Let me try again locally and get back to you though.

Comment (alexander-ding) [reply scripts/integration-test.yaml:12]: Yeah, even setting the username to `"NT AUTHORITY\\Local service"` causes commands like 
```
C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe /c & { $global:ProgressPreference = 'SilentlyContinue'; (Get-VHD -Path C:\\var\\lib\\kubelet\\plugins\\testplugin-693308.csi.io\\disk-693308.vhdx).DiskNumber }
```
to fail with permission issues.

Comment (alexander-ding) [reply scripts/integration-test.yaml:12]: And Network service fails too

**Review by mauriciopoppe — COMMENTED** [review]
/lgtm
/approve
